drivers/hmis_external_apis/app/jobs/hmis_external_apis/ac_hmis/create_referral_job.rb

```diff
@@ -163,47 +163,57 @@ def update_client_addresses(client)
             :state,
             :use,
           ),
-          AddressID: Hmis::Hud::Base.generate_uuid,
           **common_client_attrs(client),
-        }
+        )
+        new_addresses << address unless update_duplicate_record!(old_addresses, address)
       end
-      Hmis::Hud::CustomClientAddress.import!(client_address_attrs)
+
+      Hmis::Hud::CustomClientAddress.import!(new_addresses)
     end
 
+    # Add/update phones and emails
     def update_client_contacts(client)
-      # replace old phones, and emails
-      client.contact_points.destroy_all
-
-      client_phone_attrs = params[:phone_numbers].to_a.map do |values|
-        {
+      old_contact_points = client.contact_points
+      new_contact_points = []
+      params[:phone_numbers].to_a.each do |values|
+        phone = Hmis::Hud::CustomClientContactPoint.new(
           system: :phone,
           value: values[:number],
-          **values.slice(:use, :notes),
-          ContactPointID: Hmis::Hud::Base.generate_uuid,
+          use: values[:type],
+          notes: values[:notes],
           **common_client_attrs(client),
-        }
+        )
+        new_contact_points << phone unless update_duplicate_record!(old_contact_points, phone)
       end
-      Hmis::Hud::CustomClientContactPoint.import!(client_phone_attrs)
 
-      client_email_attrs = params[:email_address].to_a.map do |value|
-        {
+      params[:email_address].to_a.each do |value|
+        email = Hmis::Hud::CustomClientContactPoint.new(
           system: :email,
           value: value,
-          ContactPointID: Hmis::Hud::Base.generate_uuid,
           **common_client_attrs(client),
-        }
+        )
+        new_contact_points << email unless update_duplicate_record!(old_contact_points, email)
       end
-      Hmis::Hud::CustomClientContactPoint.import!(client_email_attrs)
+
+      Hmis::Hud::CustomClientContactPoint.import!(new_contact_points)
     end
 
     def common_client_attrs(client)
       {
         PersonalID: client.PersonalID,
-        UserID: client.UserID,
+        UserID: system_user.UserID,
```

Comment:
When LINK updates client data, it should appear like it's coming from "System User"